common/src/main/java/com/zigythebird/bendable_cuboids/mixin/playeranim/ModelMixin_playerAnim.java

```diff
@@ -1,70 +1,43 @@
 import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
 import java.util.Collection;
 import java.util.Collections;
 
 @Mixin(Model.class)
 @SuppressWarnings("UnstableApiUsage")
 public abstract class ModelMixin_playerAnim {
+    @Shadow public abstract ModelPart root();
 
-    @WrapOperation(
-            method = "renderToBuffer(Lcom/mojang/blaze3d/vertex/PoseStack;Lcom/mojang/blaze3d/vertex/VertexConsumer;III)V",
-            at = @At(
-                    value = "INVOKE",
-                    target = "Lnet/minecraft/client/model/geom/ModelPart;render(Lcom/mojang/blaze3d/vertex/PoseStack;Lcom/mojang/blaze3d/vertex/VertexConsumer;III)V"
-            )
-    )
-    public void bc$render(ModelPart instance, PoseStack poseStack, VertexConsumer buffer, int packedLight, int packedOverlay, int color, Operation<Void> original) {
-        if (this instanceof IMutableModel mutable && mutable.playerAnimLib$getAnimation() != null && mutable.playerAnimLib$getAnimation().isActive()) {
-            Collection<ModelPart> modelParts = be$getBodyParts((Model) (Object) this);
-            if (modelParts.isEmpty()) {
-                original.call(instance, poseStack, buffer, packedLight, packedOverlay, color);
-                return;
-            }
-
-            ((SodiumHelper) instance).bc$useSodiumRendering(false);
-            for (ModelPart part : modelParts) {
-                if (!((IUpperPartHelper) part).bc$isUpperPart()) {
-                    part.render(poseStack, buffer, packedLight, packedOverlay, color);
-                }
-            }
-
-            poseStack.pushPose();
-            PlayerAnimManager emoteSupplier = mutable.playerAnimLib$getAnimation();
-            PlayerAnimBone bone = new PlayerAnimBone("torso");
-            emoteSupplier.get3DTransform(bone);
-            PlayerBendHelper.applyTorsoBendToMatrix(poseStack, bone.getBend());
-
-            for (ModelPart part : modelParts) {
-                if (((IUpperPartHelper) part).bc$isUpperPart()) {
-                    part.render(poseStack, buffer, packedLight, packedOverlay, color);
-                }
-            }
-
-            poseStack.popPose();
-            ((SodiumHelper) instance).bc$useSodiumRendering(true);
-        } else {
-            original.call(instance, poseStack, buffer, packedLight, packedOverlay, color);
+    @Inject(method = "renderToBuffer(Lcom/mojang/blaze3d/vertex/PoseStack;Lcom/mojang/blaze3d/vertex/VertexConsumer;II)V", at = @At("HEAD"))
+    public void bc$render(PoseStack poseStack, VertexConsumer buffer, int packedLight, int packedOverlay, CallbackInfo ci, @Share("palActive") LocalBooleanRef palActive) {
+        palActive.set(this instanceof IMutableModel mutable && mutable.playerAnimLib$getAnimation() != null && mutable.playerAnimLib$getAnimation().isActive());
+        if (palActive.get()) {
```

Comment:
Since this class already depends on PAL, could you add a check for Sodium presence using PlayerAnimLibPlatform#isModLoaded?